uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/my-leave/apis/entitlement-api_test.js

```diff
@@ -0,0 +1,55 @@
+define([
+  'leave-absences/shared/modules/apis',
+  'leave-absences/shared/apis/entitlement-api',
+], function() {
+  'use strict';
+
+  describe('EntitlementAPI', function (){
+    var entitlementAPI;
+
+    beforeEach(module('leave-absences.apis'));
+
+    beforeEach( inject( function(_EntitlementAPI_) {
+      entitlementAPI = _EntitlementAPI_;
+    }));
+
+    it('has expected end points', function() {
+      expect(Object.keys(entitlementAPI)).toEqual(['all', 'breakdown']);
+    });
+
+    describe('gets all entitlements', function() {
+      it('response contains period id', function() {
+        entitlementAPI.all().then(function(response) {
```

Comment:
This test could be improved I think. If you have access to the mocked data, then you should know how many entitlements `.all()` will return. A valid test would be to check if `all()` truly returns *all* the entitlements